### PR TITLE
fix(msteams): Allow new install event to pass

### LINKED ISSUE #66130: Microsoft Teams Integration Problem
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Following your MS teams instructions, I have added the the Sentry bot to a team.
2. I have completed the integration with the Complete setup message I get from the bot:
image
3. Afterwards I try to set Alerts, but I get no MSTeams possibility in the create rule interface:
image

### Expected Result

I expect the possibility to select the MSTeams action in the dropdown, like shown in the documentation: https://docs.sentry.io/product/integrations/notification-incidents/msteams/

### Actual Result

Actual result:
image

### Product Area

Settings - Integrations

### Link

_No response_

### DSN

_No response_

### Version

_No response_
```[tasklist]
### Tasks
```

### LINKED ISSUE #66678: MS Team integration doesn't send Welcome message to General channel
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Go to integrations, select the Teams integration and install it by following the installation guide: https://docs.sentry.io/product/integrations/notification-incidents/msteams/

2. According to our documentation, Teams should send a Welcome message to the General channel, and users should be able to click on Complete setup there

3. The Welcome message is never sent

### Expected Result

Users should be able to complete their installation by clicking on Complete Setup

### Actual Result

Since the message is never sent, the integration cannot be successfully installed

### Product Area

Settings - Integrations

### Link

_No response_

### DSN

_No response_

### Version

_No response_

### PR DESCRIPTION
**Pull request by ykamo001**
When a new installation event comes in from the MSTeams bot, we need to allow that to go through to the Webhook event handler, instead of doing other checks and trying to get the integration or regions, as the integration does not exist yet. We were erroring because while we can technically infer an integration, it is new, and not existing.

Screenshot 2024-03-28 at 5 26 45 PM

The greater fix resolves around updating this entire middleware and changing the logic.

Resolves: https://github.com/getsentry/sentry/issues/66678

### CODE REVIEW

File: tests/sentry/integrations/msteams/webhook/test_ms_teams_webhook_mixin.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    def test_only_required_fields(self) -> None:
+        data: dict[str, Any] = {"type": "installationUpdate"}
+        assert MsTeamsWebhookMixin.is_new_integration_installation_event(data) is False
+
+    def test_additional_fields(self) -> None:
+        data: dict[str, Any] = {"type": "installationUpdate", "action": "add", "extra": "field"}
+        assert MsTeamsWebhookMixin.is_new_integration_installation_event(data) is True
+
+    def test_minimum_input(self) -> None:
```
Comment (cathteng) [inline tests/sentry/integrations/msteams/webhook/test_ms_teams_webhook_mixin.py:31]: this is the same test as on line 7?

File: tests/sentry/integrations/msteams/webhook/test_ms_teams_webhook_mixin.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    def test_invalid_event_type(self) -> None:
+        data: dict[str, Any] = {"type": "invalidType", "action": "add"}
+        assert MsTeamsWebhookMixin.is_new_integration_installation_event(data) is False
+
+    def test_invalid_action(self) -> None:
+        data: dict[str, Any] = {"type": "installationUpdate", "action": "remove"}
+        assert MsTeamsWebhookMixin.is_new_integration_installation_event(data) is False
+
+    def test_different_event_type(self) -> None:
```
Comment (cathteng) [inline tests/sentry/integrations/msteams/webhook/test_ms_teams_webhook_mixin.py:43]: this is the same test as on line 11?

**Review by cathteng — APPROVED** [review]

File: tests/sentry/integrations/msteams/webhook/test_ms_teams_webhook_mixin.py (hunk 2 above)
Comment (ykamo001) [reply tests/sentry/integrations/msteams/webhook/test_ms_teams_webhook_mixin.py:43]: ohh good catch

File: tests/sentry/integrations/msteams/webhook/test_ms_teams_webhook_mixin.py (hunk 1 above)
Comment (ykamo001) [reply tests/sentry/integrations/msteams/webhook/test_ms_teams_webhook_mixin.py:31]: yeah good catch

**Pull request merged**